negotiation/neural/models.py

```diff
@@ -596,6 +603,26 @@ def forward(self, src, tgt, lengths, dec_state=None):
             attns = None
         return decoder_outputs, attns, dec_state
 
+class NegotiationModel(NMTModel):
+
+    def __init__(self, encoder, decoder, cbow_embedder):
+        super(NegotiationModel, self).__init__(encoder, decoder)
+        self.cbow_embedder = cbow_embedder
+
+    def forward(self, sources, tgt, lengths, dec_state=None):
+        enc_final, enc_memory_bank = self.encoder(sources['enc'], lengths)
+        # item_final, item_memory_bank = self.cbow_embedder(sources['item'])
+        prev_final, prev_memory_bank = self.cbow_embedder(sources['prev'])
+        # memory_banks = [enc_memory_bank, item_memory_bank, prev_memory_bank]
+        memory_banks = [enc_memory_bank.transpose(0,1), prev_memory_bank]
```

Comment:
weird to have transpose here. do transpose at one place and write the tensor size in comment, e.g. (batch, context_len, dim)